dd-sdk-android/src/main/kotlin/com/datadog/android/error/internal/DatadogExceptionHandler.kt

```diff
@@ -73,9 +71,8 @@ internal class DatadogExceptionHandler(
         // TODO RUMM-0000 If DatadogExceptionHandler goes into dedicated module (module of 1 class
         //  only?), we have to wait for the write in some other way
         // give some time to the persistence executor service to finish its tasks
-        val coreFeature = (Datadog.globalSdkCore as? DatadogCore)?.coreFeature
-        if (coreFeature != null) {
-            val idled = (coreFeature.persistenceExecutorService as? ThreadPoolExecutor)
+        if (sdkCore is InternalSdkCore) {
```

Comment:
Let's keep it like that then, I'm not quite a fan of pulling this internal thing to the public `SdkCore`.

Regarding throwing an exception: my assumption is that if it is thrown, it will always be caught by our integration tests before leaking to the release, because it is us only who are creating the cores.

Reply:
My concern is just to avoid places in the SDK where something gets unnoticed if it is not as what we expect, this can lead to the hard debugging, especially if it happens on the client side.